pallets/phala/src/stakepoolv2.rs

```diff
@@ -0,0 +1,3177 @@
+				let new_list = vec![staker.clone()];
+				PoolContributionWhitelists::<T>::insert(&pid, &new_list);
+				Self::deposit_event(Event::<T>::PoolWhitelistCreated { pid });
+			}
+			Self::deposit_event(Event::<T>::PoolWhitelistStakerAdded { pid, staker });
+
+			Ok(())
+		}
+
+		/// Add a description to the pool
+		///
+		/// The caller must be the owner of the pool.
+		#[pallet::weight(0)]
+		pub fn set_pool_description(
+			origin: OriginFor<T>,
+			pid: u64,
+			description: DescStr,
+		) -> DispatchResult {
+			let owner = ensure_signed(origin)?;
+			let pool_info = ensure_stake_pool::<T>(pid)?;
+			ensure!(
+				pool_info.basepool.owner == owner,
+				Error::<T>::UnauthorizedPoolOwner
+			);
+			PoolDescriptions::<T>::insert(&pid, description);
+
+			Ok(())
+		}
+
+		/// Remove a staker accountid to contribution whitelist.
+		///
+		/// The caller must be the owner of the pool.
+		/// If the last staker in the whitelist is removed, the pool will return back to a normal pool that allow anyone to contribute.
+		#[pallet::weight(0)]
+		pub fn remove_staker_from_whitelist(
+			origin: OriginFor<T>,
+			pid: u64,
+			staker: T::AccountId,
+		) -> DispatchResult {
+			let owner = ensure_signed(origin)?;
+			let pool_info = ensure_stake_pool::<T>(pid)?;
+			ensure!(
+				pool_info.basepool.owner == owner,
+				Error::<T>::UnauthorizedPoolOwner
+			);
+			let mut whitelist =
+				PoolContributionWhitelists::<T>::get(&pid).ok_or(Error::<T>::NoWhitelistCreated)?;
+			ensure!(
+				whitelist.contains(&staker),
+				Error::<T>::NotInContributeWhitelist
+			);
+			whitelist.retain(|accountid| accountid != &staker);
+			if whitelist.is_empty() {
+				PoolContributionWhitelists::<T>::remove(&pid);
+				Self::deposit_event(Event::<T>::PoolWhitelistStakerRemoved {
+					pid,
+					staker: staker.clone(),
+				});
+				Self::deposit_event(Event::<T>::PoolWhitelistDeleted { pid });
+			} else {
+				PoolContributionWhitelists::<T>::insert(&pid, &whitelist);
+				Self::deposit_event(Event::<T>::PoolWhitelistStakerRemoved {
+					pid,
+					staker: staker.clone(),
+				});
+			}
+
+			Ok(())
+		}
+
+		/// Claims pool-owner's pending rewards of the sender and send to the `target`
+		///
+		/// The rewards associate to sender's "staker role" will not be claimed
+		///
+		/// Requires:
+		/// 1. The sender is a pool owner
+		#[pallet::weight(0)]
+		pub fn claim_owner_rewards(
+			origin: OriginFor<T>,
+			pid: u64,
+			target: T::AccountId,
+		) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let pool_info = ensure_stake_pool::<T>(pid)?;
+			// Add pool owner's reward if applicable
+			ensure!(
+				who == pool_info.basepool.owner,
+				Error::<T>::UnauthorizedPoolOwner
+			);
+			let rewards = pool_info.get_owner_stakes::<T>();
+			ensure!(rewards > Zero::zero(), Error::<T>::NoRewardToClaim);
+			<pallet_assets::pallet::Pallet<T> as Transfer<T::AccountId>>::transfer(
+				<T as pawnshop::Config>::PPhaAssetId::get(),
+				&pool_info.owner_account,
```

Comment:
Change `onwer_account` to `owner_reward_account`

Reply:
done